draft-irtf-cfrg-vdaf.md

```diff
@@ -823,14 +826,16 @@ Some common functionalities:
   output values.
 
 * `to_be_bytes(x: int, len: int) -> bytes` converts an integer `x` whose value
-  is in the range `[0, 2^(8*len))` to big-endian bytes.
+  is in the range `[0, 2^(8*len))` to a big-endian, `len`-byte string.
 
 * `to_le_bytes(x: int, len: int) -> bytes` converts an integer `x` whose value
-  is in the range `[0, 2^(8*len))` to little-endian bytes.
+  is in the range `[0, 2^(8*len))` to a little-endian, `len`-byte string.
 
 * `poly_eval(field: type[F], p: list[F], x: F) -> F` returns the result of
   evaluating the polynomial, `p(x)`. The coefficients of polynomials are stored
-  in lists in ascending order of degree, starting with the constant coefficient.
+  in lists in ascending order of degree, starting with the constant
+  coefficient. The `field` parameter is the class object for `F` and is used by
```

Comment:
I consider the parameter `field` is just an implementation detail, so it may not be necessary to be included in the signature of the function.

Reply:
We include it for completeness: when we use `poly_eval()` in the reference code, we must also provide the class object.

Reply:
I see two problems with this: `type[F]` 
1) notation of this expression is not defined.
2) A function with this signature `poly_eval(field: type[F], p: list[F], x: F)`
implies that the function can be invoked either:

```
poly_eval( F64, [ F64(1), F64(2)] , F64(3) )
```
or 
```
poly_eval( F32, [ F64(1), F64(2)] , F64(3) )
```

One possible solution is to mention at the beginning that all functions are parametrized with the field, or that the field is fixed for the entire scheme.

Reply:
> I see two problems with this: `type[F]`
> 
>     1. notation of this expression is not defined.

I've tried to define this, PTAL at the PR once more.

>     2. A function with this signature `poly_eval(field: type[F], p: list[F], x: F)`
>        implies that the function can be invoked either:
> 
> 
> ```
> poly_eval( F64, [ F64(1), F64(2)] , F64(3) )
> ```
> 
> or
> 
> ```
> poly_eval( F32, [ F64(1), F64(2)] , F64(3) )
> ```
> 
> One possible solution is to mention at the beginning that all functions are parametrized with the field, or that the field is fixed for the entire scheme.

Python's type checker prevents the second case: `F` is a generic parameter, so the type checker checks that `field` is the class object for `F` and that each element of the input list is an instance of `F`.

Reply:
it's ok, I needed this reference
https://docs.python.org/3/library/typing.html#typing.Type